tests/functional/runahead/01-check-default-simple.t

```diff
@@ -30,10 +30,10 @@ run_fail "${TEST_NAME}" cylc run --debug --no-detach "${SUITE_NAME}"
 #-------------------------------------------------------------------------------
 TEST_NAME="${TEST_NAME_BASE}-max-cycle"
 DB="${SUITE_RUN_DIR}/log/db"
-run_ok "${TEST_NAME}" sqlite3 "${DB}" "select max(cycle) from task_states"
-cmp_ok "${TEST_NAME}.stdout" <<'__OUT__'
-20100101T1800Z
-__OUT__
+run_ok "${TEST_NAME}" sqlite3 "${DB}" \
+    "select max(cycle) from task_states where status!='waiting'"
```

Comment:
If I understand correctly @hjoliver , the database contains tasks from the task pool + runahead pool. The latter contains only waiting tasks that have not spawned. So to correctly identify the max cycle point (which has spawned tasks), we can search just the task pool by eliminating the waiting tasks?

Reply:
Actually I'm not so sure about that explanation as I have just seen the `task_pool` table contains only the waiting tasks at the end. But regardless, the only "active" cycle points (which show up in the TUI) seem to be the ones that have at least 1 task whose state is not `waiting`.

Reply:
> If I understand correctly @hjoliver , the database contains tasks from the task pool + runahead pool.

Yes, runahead pool tasks appear as just "waiting" in the task_pool table.  On restart, all task_pool table entries are initially dropped into the runahead pool, so there's no need to distinguish the two pools in the DB.

Reply:
>  The latter [runahead pool] contains only waiting tasks that have not spawned.

I'd put it as, tasks that have been spawned but whose cycle points are still beyond the current runahead limit.  But in effect you're correct - tasks in the runahead pool can't have spawned successors yet because they haven't submitted yet (also, if a task's own next instance - i.e. same task at next cycle point - has no parents, it will be spawned when when the current instance leaves the runahead pool).

Reply:
(note that Tui [and the UI] gets its data from the "data store" not the DB, the two have a different "windowing")

Reply:
BTW the DB query above works in this instance but isn't the best way to diagnose runahead in general, but in investigating that I found that the better way is currently stymied by the "abort on stall" which doesn't check if any runahead tasks can be released before aborting.  But I'll put up a new PR for that, on top of this one.

Reply:
(Not worth putting time into my previous comment; it only affects tests. I'll resolve this discussion and simply put a code comment in this test post merge, in case the test gets flaky in the future).